docs/caa.md

```diff
@@ -0,0 +1,107 @@
+Authorities are allowed to issue certificates for your domain, you will need to
+use a DNS provider that supports setting CAA records. Check [SSLMate's CAA
+page](https://sslmate.com/labs/caa/#support) for a list of such providers. If your
+provider is listed, you can use
+[SSLMate's CAA Record Generator](https://sslmate.com/labs/caa/) to generate a
+set of CAA records listing the CAs that you would like to allow.
+
+## Where to put the record
+
+You can set CAA records on your main domain, or at any depth of subdomain.
+For instance, if you had `www.community.example.com`, you could set CAA records
+for the full name, or for `community.example.com`, or for `example.com`. CAs
+will check each version, from left to right, and stop as soon as they see any
+CAA record. So for instance, a CAA record at `community.example.com` would take
+precedence over one at `example.com`. Most people who add CAA records will want
+to add them to their registered domain (`example.com`) so that they apply to all
+subdomains.
+
+CAA validation follows CNAMEs, like all other DNS requests. If
+`www.community.example.com` is a CNAME to `web1.example.net`, the CA will first
+request CAA records for `www.community.example.com`, then seeing that there is a
+CNAME for that domain name instead of CAA records, will request CAA records for
+`web1.example.net` instead. Note that if a domain name has a CNAME record, it is
+not allowed to have any other records according to the DNS standards.
+
+The [CAA RFC](https://tools.ietf.org/html/rfc6844) specifies an additional
+behavior called "tree-climbing" that requires CAs to also check the parent
+domains of the result of CNAME resolution. Let's Encrypt does not implement tree
+climbing because it makes expressing certain CAA policies impossible. After
+[discussion on the IETF mailing
+list](https://mailarchive.ietf.org/arch/msg/spasm/vHL260X6Zb2C0VSwDwa8VZC1Kw0),
+we achieved consensus that tree-climbing in CAA is not ideal, and there's [an
+erratum](https://www.rfc-editor.org/errata/eid5065) for the CAA RFC removing it.
+That erratum still needs to be voted in by the CA/Browser Forum by September 8
+for it to be official.
+
+# CAA errors
+
+Since Let's Encrypt checks CAA records before every certificate we issue, sometimes
+we get errors even for domains that haven't set any CAA records. When we
+get an error, there's no way to tell whether we are allowed to issue for the
+affected domain, since there could be CAA records present that forbid issuance,
+but are not visible because of the error.
+
+If you receive CAA-related errors, try a few more times against our [staging
+environment](https://letsencrypt.org/docs/staging-environment/) to see if they
+are temporary or permanent. If they are permanent, you will need to file a
+support issue with your DNS provider, or switch providers. If you're not sure
+who your DNS provider is, ask your hosting provider.
```

Comment:
I think describing how to find out who your DNS provider is would complicate this doc too much.